### PR TITLE
Split global and normal logger tests

### PR DESCRIPTION
**Pull request by urso**
## What does this PR do?

Move logp global functions to global.go and selective.go, plus split up
tests. The change also adds the `nologpglobal` build tag to global.go and global_test.go.
By default all global functions are available. When compiling or
running a package its test code with `-tags=nologpglobal`, then global
logp functions are not available.

## Why is it important?

In order to better support structured logging in Beats we want to remove global logging functions. See #15699  

This PR adds a build tag that can be used to check for packages that need to be cleaned up. When running `go build -tags=nologpglobal` or `go test -tags=nologpglobal` on a selected package, the compiler will fail until the package and its transitive dependencies have been cleaned up.

## Checklist

- [x] My code follows the style guidelines of this project
- [x] I have commented my code, particularly in hard-to-understand areas
~~- [ ] I have made corresponding changes to the documentation~~
~~- [ ] I have made corresponding change to the default configuration files~~
~~- [ ] I have added tests that prove my fix is effective or that my feature works~~

## Author's Checklist

- [ ]

## Related issues

- Relates elastic/beats#15699

### CODE REVIEW

File: libbeat/logp/core_test.go [hunk 1]
```diff
@@ -53,78 +52,6 @@ func TestLogger(t *testing.T) {
...
-	}
-
-	Critical("critical")
-	logs = ObserverLogs().TakeAll()
-	if assert.Len(t, logs, 1) {
-		assert.Equal(t, zap.ErrorLevel, logs[0].Level)
-		assert.Equal(t, "", logs[0].LoggerName)
-		assert.Equal(t, "critical", logs[0].Message)
-	}
-}
```
Comment (urso) [inline libbeat/logp/core_test.go]: tests have been moved to global_test.go

File: libbeat/logp/core_test.go [hunk 2]
```diff
@@ -166,47 +93,6 @@ func TestLoggerLevel(t *testing.T) {
 	}
 }
 
-func TestRecover(t *testing.T) {
```
Comment (urso) [inline libbeat/logp/core_test.go]: tests have been moved to global_test.go

File: libbeat/logp/core_test.go [hunk 3]
```diff
@@ -166,47 +93,6 @@ func TestLoggerLevel(t *testing.T) {
...
-	panic(cause)
-}
-
-func TestHasSelector(t *testing.T) {
-	DevelopmentSetup(WithSelectors("*", "config"))
-	assert.True(t, HasSelector("config"))
-	assert.False(t, HasSelector("publish"))
-}
-
-func TestIsDebug(t *testing.T) {
```
Comment (urso) [inline libbeat/logp/core_test.go]: tests have been moved to global_test.go

File: libbeat/logp/core_test.go [hunk 4]
```diff
@@ -166,47 +93,6 @@ func TestLoggerLevel(t *testing.T) {
...
-				". Recovering, but please report this.")
-			assert.Contains(t, log.ContextMap(), "panic")
-		}
-	}()
-
-	defer Recover(recoveryExplanation)
-	panic(cause)
-}
-
-func TestHasSelector(t *testing.T) {
```
Comment (urso) [inline libbeat/logp/core_test.go]: test has been moved to selective_test.go

File: libbeat/logp/logger.go [hunk 5]
```diff
@@ -203,6 +205,14 @@ func (l *Logger) DPanicw(msg string, keysAndValues ...interface{}) {
 	l.sugar.DPanicw(msg, keysAndValues...)
 }
 
+// Recover stops a panicking goroutine and logs an Error.
+func (l *Logger) Recover(msg string) {
```
Comment (faec) [inline libbeat/logp/logger.go]: It doesn't look like this is called anywhere yet (including tests), should it be? Or is it just for use in followups?

Comment (urso) [reply libbeat/logp/logger.go]: It is for the followups.

**Pull request merged**